sdk/azcore/arm/runtime/policy_bearer_token.go

```diff
@@ -99,3 +109,26 @@ func (b *BearerTokenPolicy) onRequest(req *azpolicy.Request, authNZ func(azpolic
 func (b *BearerTokenPolicy) Do(req *azpolicy.Request) (*http.Response, error) {
 	return b.btp.Do(req)
 }
+
+// parseChallenge parses claims from an authentication challenge so a client can request a token that will satisfy
+// conditional access policies. Returns an empty string when no claims are found or claims aren't in ARM's format.
+// This function isn't universally applicable because RPs differ in challenge format and content.
+func parseChallenge(wwwAuthenticate string) string {
+	claims := ""
+	for _, param := range strings.Split(wwwAuthenticate, ",") {
+		if _, after, found := strings.Cut(param, "claims="); found {
+			if claims != "" {
+				// The header contains multiple challenges, at least two of which specify claims. The specs allow this
+				// but it's unclear what a client should do in this case and there's as yet no concrete example of it.
+				return ""
+			}
+			// trim stuff that would get an error from RawURLEncoding; claims may or may not be padded
+			claims = strings.Trim(after, `\"=`)
+			// we don't return the error because when not nil it's something unhelpful like "illegal base64 data at input byte 42"
+			if b, err := base64.RawURLEncoding.DecodeString(claims); err == nil {
```

Comment:
If this does fail, is the assumption that the claim isn't encoded?

Reply:
Yes. That would be surprising because all claims from ARM should be base64 encoded.

After reading this again, I made some changes to distinguish failure to parse claims from absence of claims and enable the policy to return a `ResponseError` in the latter case, as clients do today. (ARM includes claims in the challenge only when it rejects a request due to a conditional access policy.)